src/octopeer-github/test/main/Options/OptionsTest.ts

```diff
@@ -166,14 +166,48 @@ describe("The Options class", function() {
             // nothing found, so approve.
             return true;
         }
-        
-        expect(
-            recursivelyTopologicalyRemove(
-                $.map(
-                    Options.DEPENDENCIES,
-                    (value, index) => [[index, value]]
+
+        let a = "a", b = "b", c = "c", d = "d", e = "e", f = "f";
+
+        // now for testing the functions used in the test
+        it("should be so that a empty mapping gets approved", function() {
+            expect(recursivelyTopologicalyRemove([])).toBe(true);
+        });
+
+        it("should be so that a non circular mapping gets approved", function() {
+            expect(recursivelyTopologicalyRemove([
+                [a, b],
+                [b, c],
+                [c, d],
+                [e, f],
+            ])).toBe(true);
+        });
+
+        it("should be so that a circular mapping gets rejected", function() {
+            expect(recursivelyTopologicalyRemove([
+                [a, b],
+                [b, c],
+                [c, d],
+                [c, a],
+                [e, f],
+            ])).toBe(false);
+        });
+
+        it("should be so that a simple circular mapping gets rejected", function() {
+            expect(recursivelyTopologicalyRemove([
+                [a, a],
+            ])).toBe(false);
+        });
+
+        it("should be the case that there are no circular dependancies specified in the Options", function() {
+            expect(
+                recursivelyTopologicalyRemove(
+                    $.map(
+                        Options.DEPENDENCIES,
+                        (value, index) => [[index, value]]
+                    )
```

Comment:
Can `$.map(Options.DEPENDENCIES, (value, index) => [index, value])` be one line?